new/server-stats-endpoint.md

```diff
@@ -0,0 +1,114 @@
+
+### Supported Metric Primitives
+The endpoints can provide metrics of following types, as defined by [Prometheus](https://prometheus.io/docs/concepts/metric_types/):
+- Counter - Monotonic cumulative metric. For example, number of requests served.
+- Gauge - Numeric measurement at a given point in time. For example, number of inflight requests.
+- Histogram - Distributes values into buckets. For example, response times.
+- Summary - Provides summary statistics for a value in a sliding window.
+
+### Response Format
+#### Why Use the Prometheus Format?
+The format is already defined and provides metric types that are known to work with at least
+one monitoring system. This format also gives us the ability to describe each metric presented,
+so the end user would know what it means. The format itself is human readable and easy to parse,
+making it easy to both get started with and write tooling around.
+
+Using this format directly also could allow us to use metric exporters already written for other co-located
+services such as PostgreSQL, RabbitMQ, and Solr.
+
+#### Why Not?
+This would introduce an inconsistency in that the current API uses JSON and this one route would not. We
+could just as well create a JSON specification with information such as type, description, metric name, etc.
+
+
+### Services
+At a minimum, the Chef Server frontend service would need to provide stats. For example, `bookshelf`, `oc-id`,
+`oc_bifrost`, and `erchef` should provide metrics as those are always colocated. Since these run together, these could also be grouped into a single metrics endpoint(`_stats/chef-server`) to reduce the number of calls needed to gather the metrics.
+
+If Chef Server is configured to provide services such as RabbitMQ, Solr, Elasticsearch, PostgreSQL, Manage, Reporting, etc, an endpoint should be provided for each service.
+
+#### Service List
+Since different implementations or configurations of Chef Server could have different dependencies, 
+`GET /_stats` will provide a list of services.
+
+Example response:
+
+```
+{
+  "services": [
+    "chef-server",
+    "solr",
+    "rabbitmq"
+  ]
+}
+```
+
+Each one of these services must be serviceable by `_stats/$service`, for example `_stats/solr`.
+
+## Yet Another Way to Get Metrics
+This introduces yet another way to get metrics for the Chef Server. All metrics that were previously
+available through statsd or Folsom will be available from the stats endpoint. This means that other
+methods should be deprecated and removed in the next major version bump of Chef Server. Those who
+wish to push to Graphite may still do so by polling. Doing the same for statsd may no longer make sense
+as the stats endpoint will have already aggregated the metric data. That being said, those who wish to
+still use statsd can get the data from the request logs.
+
+## Unanswered Questions
+
+- What about chef-backend?
+- Do we need or should we have authn/authz?
+- Versioning
```

Comment:
If this is part of the API we could re-use the API versioning headers.  However, I think part of the contract should be that clients should not expect any particular stats or services since we want to make it easy to add statistics as the system changes.